src/hb-ot-hdmx-table.hh

```diff
@@ -196,9 +196,18 @@ struct hdmx
   inline bool sanitize (hb_sanitize_context_t *c) const
   {
     TRACE_SANITIZE (this);
-    return_trace (c->check_struct (this) && version == 0 &&
-		  !_hb_unsigned_int_mul_overflows (num_records, size_device_record) &&
-		  c->check_range (this, get_size()));
+    if (!c->check_struct (this) || version != 0 ||
+	_hb_unsigned_int_mul_overflows (num_records, size_device_record) ||
+	!c->check_range (this, get_size()))
+    {
+      return_trace (false);
+    }
+    for (unsigned int i = 0; i < this->num_records; i++)
```

Comment:
Since it's a compact array, I suggest you just add one check_range. Indeed, that's what you were doing. I think what you were missing was a check that `size_device_record >= DeviceRecord::min_size`. No?

Reply:
You're correct. I've removed the for loop and just added size_device_record >= DeviceRecord::min_size.